version: '3.8'

networks:
  proxy:
    external: true

volumes:
  portainer_data:

services:
  traefik:
    image: 'traefik:latest'
    container_name: traefik
    restart: unless-stopped
    security_opt:
      - 'no-new-privileges:true'
    networks:
      - proxy
    ports:
      - '80:80'
      - '443:443'
    volumes:
      - '/etc/localtime:/etc/localtime:ro'
      - '/var/run/docker.sock:/var/run/docker.sock:ro'
      - './traefik.yml:/traefik.yml:ro'
      - './acme.json:/acme.json'
      - './configurations:/configurations'
    labels:
      - traefik.enable=true
      - traefik.docker.network=proxy
      - traefik.http.routers.traefik-secure.entrypoints=websecure
      - traefik.http.routers.traefik-secure.rule=Host(`traefik.example.com`)
      - traefik.http.routers.traefik-secure.service=api@internal
      - traefik.http.routers.traefik-secure.middlewares=user-auth@file

  portainer:
    image: 'portainer/portainer-ce:latest'
    container_name: portainer
    restart: unless-stopped
    security_opt:
      - no-new-privileges:true
    networks:
      - proxy
    volumes:
      - /etc/localtime:/etc/localtime:ro
      - /var/run/docker.sock:/var/run/docker.sock
      - ./volumes/portainer:/data
    labels:
      - traefik.enable=true
      - traefik.docker.network=proxy
      - traefik.http.routers.portainer-secure.entrypoints=websecure
      - traefik.http.routers.portainer-secure.rule=Host(`portainer.example.com`)
      - traefik.http.routers.portainer-secure.service=portainer
      - traefik.http.services.portainer.loadbalancer.server.port=9000
